src/components/SearchFunction.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Button from './Button';
+import SearchState from './SearchState.js'
+
+class SearchFunction extends React.Component {
```

Comment:
Although you are breaking the components down pretty well, I'd say using names like `function` and `state` for components are confusing. It is better to think of components as visual components, so for example SearchForm and SearchResults would be better.